src/libs/tools/src/plugindatabase.cpp

```diff
@@ -379,6 +380,108 @@ std::vector<PluginSpec> ModulesPluginDatabase::lookupAllProvides (std::string co
 }
 
 
+class PluginVariantsDatabase::Impl
+{
+public:
+	Impl ()
+	{
+	}
+	~Impl ()
+	{
+	}
+	Modules modules;
```

Comment:
I am not sure what you tried and what the error was. Please update the PR with the issue you have (you can rebase the last commit later)

> But how to add the member without inheriting the class?

You have two different `impl` and can access both as needed.

> Can I just add the KeySet pluginconf to ModulesPluginDatabase::Impl?

From ABI point of view: yes. But it would be strange because ModulesPluginDatabase does not need it.